compiler/qsc_frontend/src/typeck.rs

```diff
@@ -39,7 +39,7 @@ pub(super) struct Error(ErrorKind);
 enum ErrorKind {
     #[error("expected {0}, found {1}")]
     #[diagnostic(code("Qsc.TypeCk.TyMismatch"))]
-    TyMismatch(Ty, Ty, #[label] Span),
+    TyMismatch(String, String, #[label] Span),
```

Comment:
I like this. Strings instead of HIR objects in the error makes a lot of sense to me!

Reply:
They are smaller in size too 😊